golang/x/relay/keeper/querier.go

```diff
@@ -234,3 +236,35 @@ func queryGetRequest(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) (res
 	}
 	return res, nil
 }
+
+func queryCheckRequests(ctx sdk.Context, req abci.RequestQuery, keeper Keeper) (res []byte, err sdk.Error) {
+	var params types.QueryParamsCheckRequests
+	var errMsg string
+	valid := true
+
+	unmarshallErr := types.ModuleCdc.UnmarshalJSON(req.Data, &params)
+	if unmarshallErr != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("failed to parse params: %s", unmarshallErr))
+	}
+
+	// This calls the keeper with the parsed arguments, and gets an answer
+	resErr := keeper.checkRequestsFilled(ctx, params.Filled)
+	if resErr != nil {
+		valid = false
+		errMsg = resErr.Error()
+	}
+
+	// Now we format the answer as a response
+	response := types.QueryResCheckRequests{
+		Params:       params,
+		Valid:        valid,
+		ErrorMessage: errMsg,
+	}
+
+	// And we serialize that response as JSON
+	res, marshalErr := codec.MarshalJSONIndent(keeper.cdc, response)
+	if marshalErr != nil {
+		return []byte{}, types.ErrMarshalJSON(types.DefaultCodespace)
+	}
+	return res, nil
```

Comment:
hmmm...I think I'll populate the `err` field as well in the event `checkRequestsFilled` returns an error. There are many different errors that could have been returned from `checkRequestsFilled` including a badly formatted request...

Reply:
not sure if that goes against golang conventions though...maybe one should always be nil?

Reply:
I think thoughts like these left me confused as to which approach to take

Reply:
personally I would never populate both. I'd always leave 1 as the default value 0 or nil or whatever. Either it worked or it didn't, ya know?

In this case, returning an error will signal to the REST and CLI agents that they should treat it as though the request errored. which is not what we want. We want the REST and CLI to treat it a `checkRequestsFilled` error as normal operation (and JSON errors, etc as actual errors)

Reply:
makes sense. that means that this transaction will be slightly different than other ones in the sense that an invalid Proof param will result in a nil error, but the response will notify them that validation failed and the Proof was formatted poorly.

Reply:
right. essentially this is using eth_call to view-call a non-view function in solidity. It'll complete without an error and tell you that the execution will error.